java/src/jmri/jmrix/can/AbstractCanTrafficController.java

```diff
@@ -301,11 +305,10 @@ public void distributeOneReply(CanReply msg, AbstractMRListener mLastSender) {
         // which includes the communications monitor
         // return a notification via the Swing event queue to ensure proper thread
         Runnable r = newRcvNotifier(msg, mLastSender, this);
-        try {
-            javax.swing.SwingUtilities.invokeAndWait(r);
-        } catch (java.lang.reflect.InvocationTargetException | InterruptedException | RuntimeException e) {
-            log.error("Unexpected exception in invokeAndWait:" + e);
-            e.printStackTrace();
+        if (getSynchronizeRx()) {
+            ThreadingUtil.runOnLayout(r::run);
```

Comment:
If you want this in the GUI thread, run it in methods that ensure it is (use runOnGUI*). This runs on the Layout thread, which is currently the same as the GUI thread, but there is no code contract to ensure that does not change.

Reply:
you told me to run it on the layout thread. I'm not happy to deviate from what AbstractMRTRafficController is doing, but there is no contract described there on where the layout input events are running. I think it's a mistake to make such a wide reaching design decision in this PR and then implement it only partially.